packages/fxa-payments-server/src/routes/Subscriptions/index.tsx

```diff
@@ -1,130 +1,345 @@
+        <AlertBar className="alert alertPending">
+          <span>
+            Updating billing information...
+          </span>
+        </AlertBar>
+      )}
+
+      {updatePaymentStatus.error && (
+        <DialogMessage className="dialog-error" onDismiss={resetUpdatePayment}>
+          <h4>Updating billing information failed</h4>
+          <p>{updatePaymentStatus.error.message}</p>
+        </DialogMessage>
+      )}
+
+      {reactivateSubscriptionStatus.error && (
+        <DialogMessage className="dialog-error" onDismiss={resetReactivateSubscription}>
+          <h4>Reactivating subscription failed</h4>
+          <p>{reactivateSubscriptionStatus.error.message}</p>
+        </DialogMessage>
+      )}
+
+      {profile.result && ( <ProfileBanner profile={profile.result} /> )}
+
+      <div className="child-views">
+
+        <div className="settings-child-view support">
+
+          <div className="settings-unit">
+
+            <div className="settings-unit-stub">
+              <header className="settings-unit-summary">
+                <h2 className="settings-unit-title">Subscriptions</h2>
+              </header>
+              <button className="settings-button secondary-button settings-unit-toggle" onClick={onSupportClick}>
+                <span className="change-button">Contact Support</span>
+              </button>
+            </div>
+
+          </div>
+
+          {customerSubscriptions.length === 0 && <>
+            {/*
+              TODO: User probably shouldn't end up here, since the "Manage"
+              button should be unavailable from FxA settings.
+              https://github.com/mozilla/fxa/issues/1078
+            */}
+            <div className="settings-unit">
+              <div className="subscription">
+                <p>No subscriptions available.</p> 
+              </div>
+            </div>
+          </>}
+
+          {customerSubscriptions.map((customerSubscription, idx) =>
+            <SubscriptionItem key={idx} {...{
+              accessToken,
+              customer,
+              updatePayment,
+              resetUpdatePayment,
+              updatePaymentStatus,
+              cancelSubscription,
+              reactivateSubscription,
+              customerSubscription,
+              plan:
+                planForId(customerSubscription.plan_id, plans),
+              subscription:
+                subscriptionForId(customerSubscription.subscription_id, subscriptions),
+            }} />
+          )}
+
+        </div>
+
+      </div>
 
-      {customerSubscriptions.map((subscription, idx) =>
-        <Subscription key={idx} {...{ accessToken, cancelSubscription, subscription }} />)}
     </div>
   );
 };
 
+const customerSubscriptionForId = (
+  subscriptionId: string,
+  customerSubscriptions: Array<CustomerSubscription>,
+): CustomerSubscription | null => {
+  return customerSubscriptions.filter(subscription =>
+    subscription.subscription_id === subscriptionId)[0];
```

Comment:
Is this just assuming that `[][0]` will return null as a fallback if it's not found? (Though I guess it would be undefined)

Reply:
Yeah, aiming for a falsy fallback here I think